### PR TITLE
fix(elements): Show warning on used, but not activated verifications

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Add a development-only warning for cases when a user renders a `<Strategy>` component that isn't activated for their Clerk instance. As this can be intended behavior (e.g. build out a full example and let user enable/disable stuff solely in the dashboard) the warning can safely be ignored if necessary.

Since I touched on warnings/errors in general, I improved also some other existing ones.

Example of now more prominent error:

CleanShot 2024-06-20 at 15 39 45

Example of navigation event:

CleanShot 2024-06-21 at 08 41 26

Example of new warning:

CleanShot 2024-06-21 at 10 33 39

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/elements/src/internals/machines/sign-in/verification.machine.ts [hunk 1]
```diff
@@ -92,49 +92,66 @@ const SignInVerificationMachine = setup({
...
-      ) {
-        console.warn(
-          `Clerk: Your instance is configured to support these strategies: ${clerk.client.signIn.supportedFirstFactors
-            .map((factor: any) => factor.strategy)
-            .join(', ')}, but the rendered strategies are: ${Array.from(context.registeredStrategies).join(
-            ', ',
-          )}. Before deploying your app, make sure to render a <Strategy> component for each supported strategy. For more information, visit the documentation: https://clerk.com/docs/elements/reference/sign-in#strategy`,
-        );
-      }
+      // Only show these warnings in development!
```
Comment (LekoArts) [inline packages/elements/src/internals/machines/sign-in/verification.machine.ts:95]: I think it makes sense to show all of them only during development, or what do y'all think?

File: packages/elements/src/internals/machines/sign-in/verification.machine.ts [hunk 2]
```diff
@@ -92,49 +92,66 @@ const SignInVerificationMachine = setup({
...
+      if (process.env.NODE_ENV === 'development') {
+        if (
+          !clerk.client.signIn.supportedFirstFactors.every(factor =>
+            // TODO: These "as SignInFactor" assertions are incorrect, factor.strategy is SignInFactor['strategy']. This needs to be fixed together with SignInVerificationStrategyRegisterEvent and SignInStrategy React component
```
Comment (LekoArts) [inline packages/elements/src/internals/machines/sign-in/verification.machine.ts:99]: Created SDK-1845 for those TODOs

File: packages/elements/src/internals/machines/sign-in/verification.machine.ts [hunk 3]
```diff
@@ -92,49 +92,66 @@ const SignInVerificationMachine = setup({
...
+            !clerk.client.signIn.supportedFirstFactors.some(
+              supported => (supported.strategy as unknown as SignInFactor) === strategy,
+            ),
+        ) as unknown as Array<SignInFactor['strategy']>;
+
+        if (strategiesUsedButNotActivated.length > 0) {
+          console.warn(
+            `Clerk: These rendered strategies are not configured for your instance: ${strategiesUsedButNotActivated.join(', ')}. If this is unexpected, make sure to enable them in your Clerk dashboard: https://dashboard.clerk.com/last-active?path=/user-authentication/email-phone-username`,
+          );
+        }
```
Comment (LekoArts) [inline packages/elements/src/internals/machines/sign-in/verification.machine.ts:140]: This is the new behavior. I think a diff is a nice DX touch we could in theory also use for the other warnings. Then you don't have to compare two lists

File: packages/elements/src/internals/machines/sign-in/verification.machine.ts (hunk 1 above)
Comment (tmilewski) [reply packages/elements/src/internals/machines/sign-in/verification.machine.ts:95]: @LekoArts I'm torn on this one as a prod config can differ from dev.

**Review by tmilewski — APPROVED** [review]

**Pull request merged**